docs/dev/azure-config-credentials-injector.md

```diff
@@ -0,0 +1,40 @@
+# Azure credentials injector
+
+## Motivation
+
+There are various auth mechanisms for cloud-controller-manager(CCM) and cloud-node-manager(CNM) within azure.
+
+Namely: 
+  * [Using](https://github.com/openshift/cloud-provider-azure/blob/master/pkg/auth/azure_auth.go#L95) the [managed identity extension](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview) 
+    * this is what openshift leveraging for public Azure installations at the moment (4.8)
+  * Store part of cloud config in a secret, configure CCM accordingly (point it to appropriate secret key)
+    * Secret should contain json object with represents a part of [cloud-config](https://kubernetes-sigs.github.io/cloud-provider-azure/install/configs/) 
+    * It would be merged with `cloud-config` during CCM startup procedure
+    * CNM does not support such merge behaviour at the moment
+  * Pass credentials via cloud-config directly as corresponding parameters
+  
+Due to existing secret format described in [OCP documentation](https://docs.openshift.com/container-platform/4.8/installing/installing_azure/manually-creating-iam-azure.html#admin-credentials-root-secret-formats_manually-creating-iam-azure) which is not directly applicable for CCM/CNM,
+`azure-config-credentials-injector` was introduced.
+
+## azure-config-credentials-injector 
+
+Due to no support of [managed identity extension](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview) on Azure StackHub we need to pass credentials to CCM/CNM somehow.
+There are couple of issues with this process at the moment:
+* Secret with credentials created by [cloud-credential-operator](https://github.com/openshift/cloud-credential-operator) can not be used directly due to different format
+* CNM does not support merging behaviour (part of cloud config in a `secret` resource in other words)
+
+
+As temporary solution of necessity to pass credentials to CCM/CNM on Azure StackHub platform separate binary within CCCMO operator image was introduced.
```

Comment:
We (storage team) are interested in this too. How *temporary solution* is it? Can we depend on the injector image to be maintained as long as we need to? We're willing to help with the maintenance, of course.

Reply:
I think probably for a few releases at least. Our hope was that we would be able to in the future introduce the support needed to merge the credentials within the cloud provider code itself, so that for example, it can override the credential config from env vars.

I'm not sure we've even started a conversation upstream about this yet so this may not actually be accepted as a reasonable approach